Objects/longobject.c

```diff
@@ -1199,17 +1219,51 @@ PyLong_AsNativeBytes(PyObject* vv, void* buffer, Py_ssize_t n, int endianness)
             _PyLong_AsByteArray(v, buffer, (size_t)n, little_endian, 1, 0);
         }
 
-        // More efficient calculation for number of bytes required?
+        /* Calculates the number of bits required for the *absolute* value
+         * of v. This does not take sign into account, only magnitude. */
         size_t nb = _PyLong_NumBits((PyObject *)v);
```

Comment:
Turns out `_PyLong_NumBits` can, theoretically, fail. (But I don't think there's a good way to test such a huge allocation.)

```suggestion
        size_t nb = _PyLong_NumBits((PyObject *)v);
        if (nb == (size_t)-1 && PyErr_Occurred()) {
            if (do_decref) {
                Py_DECREF(v);
            }
            return -1;
        }
```